docs/apache-airflow-providers/index.rst

```diff
@@ -87,9 +88,24 @@ The capabilities are:
   connections defined by the provider. See :doc:`apache-airflow:howto/connection` for a description of
   connection and what capabilities of custom connection you can define.
 
+Custom provider packages
+''''''''''''''''''''''''
+
+However, there is more. You can develop your own providers. This is a bit involved, but your custom operators,
+hooks, sensors, transfer operators can be packaged together in a standard airflow package and installed
+using the same mechanisms. Moreover they can also use the same mechanisms to extend the Airflow Core with
+custom connections and extra operator links as described in the previous chapter. See
+[Developing custom providers](https://github.com/apache/airflow/blob/master/DEVELOPING_CUSTOM_PROVIDERS.rst)
```

Comment:
where is this file? -- I think currently it is in a separate section in Contributing.rst

Reply:
Right. It was to short for one file and I merged it in contributing, but did not update the reference :)

Reply:
Actually I remove it. Because it is described in the next chapter. This is user-facing documentation if you want to develop your own, custom provider (with a nice FAQ entry below added by one of the users who already developed their own provider and share the learnings). The CONTRIBUTING doc is only for those who would like to develop the community provider (with documentation, provider.yaml etc).

Reply:
I updated the documentation a bit, adding more requirements for custom provider documentation, mentioning testing/documentaion requirements.